### PR TITLE
restore disabled FileSystemTests

### PR DESCRIPTION
**Pull request by tomerd**
motivation: better test coverage

changes:
* refactor the common part of testInMemoryFileSystemFileLock, testLocalFileSystemFileLock, testRerootedFileSystemViewFileLock
* resove the potential reader race by making sure file have content prior to starting the threads
* enable the above disabled tests in FileSystemTests

rdar://71560894

### CODE REVIEW

**Comment by tomerd** [comment]
@swift-ci please test

File: Tests/TSCBasicTests/FileSystemTests.swift [hunk 1]
```diff
@@ -884,22 +783,31 @@ class FileSystemTests: XCTestCase {
         let fileB = path.appending(component: "fileB")
         let lockFile = path.appending(component: "lockfile")
 
+        try _testFileSystemFileLock(fileSystem: fs, fileA: fileA, fileB: fileB, lockFile: lockFile)
+    }
+
+    private func _testFileSystemFileLock(fileSystem fs: FileSystem, fileA: AbsolutePath, fileB: AbsolutePath, lockFile: AbsolutePath) throws {
+
+        // write initial value, since reader may start before writers and files would not exist
+        try fs.writeFileContents(fileA, bytes: [0])
```
Comment (benlangmuir) [inline Tests/TSCBasicTests/FileSystemTests.swift:792]: This is writing a null byte, so the code below that tries to read an integer as a string from the file will still fail.  I guess it's "okay" in the sense that it will then hit the `?? 0`, but it seems kind of indirect and hard to follow, especially since it's easy to misread writing 0 as being equivalent to writing "0".  Is there a reason not to write "0" to the file instead of "\0" and then simplify the logic below to force the optional Int?

File: Tests/TSCBasicTests/FileSystemTests.swift [hunk 2]
```diff
@@ -884,22 +783,31 @@ class FileSystemTests: XCTestCase {
...
                     // Sum and write back to file.
-                    try! fs.writeFileContents(fileA, bytes: ByteString(encodingAsUTF8: String(valueA + 1)))
+                    try fs.writeFileContents(fileA, bytes: ByteString(encodingAsUTF8: String(valueA + 1)))
 
                     Thread.yield()
 
-                    // Get thr current contents of the file if any.
+                    // Get the current contents of the file if any.
                     let valueB: Int
                     if fs.exists(fileB) {
```
Comment (benlangmuir) [inline Tests/TSCBasicTests/FileSystemTests.swift:812]: Always true now

File: Tests/TSCBasicTests/FileSystemTests.swift (hunk 1 above)
Comment (tomerd) [reply Tests/TSCBasicTests/FileSystemTests.swift:792]: good point, I can revise

**Comment by tomerd** [comment]
@swift-ci please test

**Review by neonichu — APPROVED** [review]
